### DISCUSSION TITLE
Bounty for Fixes/Features - Updated

### DISCUSSION THREAD

**Discussion by Terracotta1** [xbapps/xbvr#637] [Ideas]
I'll pay $100 per section for the three sections below. I realize that's not a lot of $ for how long some of these will take but maybe it'll be a nice incentive for someone already working on the project.

-Make the DeoVR integration behave more like paid websites where you can search for names/websites/categories.
-Also add the ability to sort alphabetically, randomly, and by the files' Date Modified attribute instead of only being able to sort by scene date.

-When doing manual matching, 
--add a Play button in the matching window so you can see what the file is if the name isn't helpful.
--add a Scrape button to each result row that would trigger the "Run this scraper" function for the website in that row, like for when you know that site has that scene but it's so new that it's not yet in your scrape results.
--add the ability to assign a name and image and any other required information to a file that can't be matched to any scraped scene, like for scenes that have been removed from websites.
--make the position of the Assign buttons in the matching window stationary across pages.
--make it so matching a file doesn't revert the file list to page 1.

-Option to set a Score-based threshold where automatic matching will work even if it's not 100% certain.
-Option to change the frequency at which folders are scanned for new files, or even to do real-time scanning.
-Option to bulk rename matched files to what they are expected to be, to tidy up filenames and to make automatic matching work in the future in the event the database is lost or corrupted.
-Ability to un-match files without having to rename and re-scan.
-Ability to search for scenes that have more than 1 matched file, to be able to find duplicates.
-Make the "scrape" folder be part of the config directory instead of the executable so I can place fixed or new scraper files in there instead of having to recompile every time.

**Comment by theRealKLH** [comment]
some of these are already done, some were coded but not (yet) integrated (open PRs), some were discussed on Discord, some you WOULDN'T want (ie real-time scanning). Some you have the wrong understanding (ie scores not involved in "automatching" & you don't want this thing guessing incorrectly about what scene your vid actually is-this was just discussed last couple days on Discord)

**Comment by Terracotta1** [comment]
Hi,

Thanks for the reply. I see there's already a way to skip over files when manually matching (not sure how I missed that) and I see un-matching and finding duplicates were both added in open pull requests, so I'll wait for those to be merged. I gave it a go myself and have been able to implement other open PRs but can't figure out how to do those two specifically since the affected files have been altered by closed PRs in the months since they were submitted.

For the real-time scanning and the more casual auto matching, you're right that I probably don't fully understand how they work, but I do know that other similar programs can do those things, like Plex w/ scanning and Sonarr w/ matching, so surely they're possible in a general sense even if it involves rewriting the code?

**Comment by houserwx** [comment]
> I'll pay $100 per section for the three sections below. I realize that's not a lot of $ for how long some of these will take but maybe it'll be a nice incentive for someone already working on the project.
> 
> -Make the DeoVR integration behave more like paid websites where you can search for names/websites/categories. -Also add the ability to sort alphabetically, randomly, and by the files' Date Modified attribute instead of only being able to sort by scene date.
> 
> -When doing manual matching, --add a Play button in the matching window so you can see what the file is if the name isn't helpful. --add a Scrape button to each result row that would trigger the "Run this scraper" function for the website in that row, like for when you know that site has that scene but it's so new that it's not yet in your scrape results. --add the ability to assign a name and image and any other required information to a file that can't be matched to any scraped scene, like for scenes that have been removed from websites. --make the position of the Assign buttons in the matching window stationary across pages. --make it so matching a file doesn't revert the file list to page 1.
> 
> -Option to set a Score-based threshold where automatic matching will work even if it's not 100% certain. -Option to change the frequency at which folders are scanned for new files, or even to do real-time scanning. -Option to bulk rename matched files to what they are expected to be, to tidy up filenames and to make automatic matching work in the future in the event the database is lost or corrupted. -Ability to un-match files without having to rename and re-scan. -Ability to search for scenes that have more than 1 matched file, to be able to find duplicates. -Make the "scrape" folder be part of the config directory instead of the executable so I can place fixed or new scraper files in there instead of having to recompile every time.

I have the following:
- Move or delete all but the largest file from duplicate matches for a scene (currently I have them move to a /Dupes folder)
- Score threshold matching, as well as sub-string match (first it looks for the expected embedded in the name to allow for rarbg etc, then it accepts top match over 100% currently, but that gets a decent amount of false positives, so pushing that higher might work)
- Bulk rename - if single match does not match the first file name entry on the first entry of scene, it updates it.
- I can add anything you like to the db/filemanagement through the web UI.  Let me know how you want to get in touch.
- Move all unmatched files to an "unmatched" folder / move all "matched" files to a "matched" folder
-
